enterprise_gateway/services/kernels/handlers.py

```diff
@@ -117,10 +127,16 @@ async def get(self):
         tornado.web.HTTPError
             403 Forbidden if kernel listing is disabled
         """
-        if not self.settings.get("eg_list_kernels"):
+
+        tenant_id_filter = self.request.query_arguments.get("tenant_id") or UNIVERSAL_TENANT_ID
+        if isinstance(tenant_id_filter, list):
+            tenant_id_filter = tenant_id_filter[0].decode("utf-8")
+        if not self.settings.get("eg_list_kernels") and tenant_id_filter == UNIVERSAL_TENANT_ID:
```

Comment:
why do we have `and tenant_id_filter == UNIVERSAL_TENANT_ID:` check to this?

Reply:
Today, EG requires that "list kernels" functionality be enabled explicitly - default it off.  This would merely allow those configurations that use their own tenant_ids to not have to set that configurable.  It's purely for convenience and I'd be okay not including this little "easter egg".